File: src/utils/commands/flow.ts
import { invoke } from '@tauri-apps/api/core';

export async function listFlows(): Promise<string[]> {
  return invoke('list_flows');
}

export async function getFlow(name: string): Promise<string> {
  return invoke('get_flow', { name });
}

export async function executeFlow(name: string, params: string): Promise<void> {
  return invoke('execute_flow', { name, params });
}

File: src-tauri/src/commands/flow.rs
use crate::orchestrator::types::Flow;

#[tauri::command]
pub async fn list_flows() -> Result<Vec<String>, String> {
    // TODO: Read flow files from workflows repo
    Ok(vec![])
}

#[tauri::command]
pub async fn get_flow(name: String) -> Result<String, String> {
    // TODO: Read and return flow JSON
    Ok("{}".to_string())
}

#[tauri::command]
pub async fn execute_flow(name: String, params: String) -> Result<(), String> {
    // TODO: Start flow orchestration
    Ok(())
}
